src/components/tab-panel/tab-panel.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+    Component,
+    Element,
+    h,
+    Prop,
+    EventEmitter,
+    Event,
+    Host,
+} from '@stencil/core';
+import { Tab } from '../tab-bar/tab.types';
+
+/**
+ * @slot - Content to put inside the `tab-panel`. Each slotted element must have the `id` attribute equal to the id of the tab it belongs to.
+ */
+@Component({
+    tag: 'limel-tab-panel',
+    shadow: true,
+})
+export class TabPanel {
+    @Prop({ mutable: true })
+    public tabs: Tab[] = [];
+
+    @Element()
+    private host: HTMLElement;
+
+    @Event()
+    protected changeTab: EventEmitter<Tab>;
+
+    private slotElements: HTMLElement[];
+
+    constructor() {
+        this.handleChangeTabs = this.handleChangeTabs.bind(this);
+        this.setSlotElements = this.setSlotElements.bind(this);
+        this.setTabStatus = this.setTabStatus.bind(this);
+    }
+
+    protected componentDidLoad() {
+        const slot = this.getSlot();
+        slot.addEventListener('slotchange', this.setSlotElements);
+        this.setSlotElements();
+        this.tabs.forEach(this.setTabStatus);
+    }
+
+    protected componentDidUnload() {
+        const slot = this.getSlot();
+        slot.removeEventListener('slotchange', this.setSlotElements);
+    }
+
+    public render() {
+        return (
+            <Host onChangeTab={this.handleChangeTabs}>
+                <div class="tab-panel">
+                    <limel-tab-bar tabs={this.tabs} />
+                    <div class="tab-content">
+                        <slot />
+                    </div>
+                </div>
+            </Host>
+        );
+    }
+
+    private setSlotElements() {
+        const slot = this.getSlot();
+        this.slotElements = [].slice.call(slot.assignedElements());
+    }
+
+    private setTabStatus(tab: Tab) {
+        const element = this.slotElements.find(e => e.id === tab.id);
+        if (tab.active) {
+            element.style.display = '';
+        } else {
+            element.style.display = 'none';
+        }
+
+        element['tab'] = tab; // tslint:disable-line:no-string-literal
```

Comment:
It feels quite nice that this is done automatically, but I'm not 100% sure. If it's not done here, the user has to send in the tab manually for each component

Reply:
This is used by `tab-panel-content.tsx` right?

Off the top of my head, I can't come up with any reason we shouldn't do this automatically, but please do share your concerns, there are probably things I haven't thought of!

Reply:
Yes, any content inside the tab-panel will receive the corresponding tab configuration as a `Prop() tab`. The component needs to declare a property named `tab` or it won't work, so it's kind of implicit. If the component already has another prop named `tab`, we will override it with this line of code.
We can make it more explicit by not doing this and forcing the consumer to send in the tab manually when using the tab-panel. It might make it slightly harder for us in the context of the webclient when we want to dynamically render custom web-components from consultants. In that case, we might have to send it in on the context or something like that :thinking:

Reply:
Ah, ok. I understand your doubts then, but yes, it seems pretty nice to have it done automatically. I guess we could define an interface `TabPanel` which defines that it must have a `tab` property of type `Tab`? That wouldn't fix the "if it already exists we'll overwrite it"-part, but it ought to make it explicit enough at least?

Reply:
Yeah maybe :thinking: 
There's no way to enforce it though, so I wonder how useful it would be :thinking: 
The probability of placing a component in the tab-panel with a property named `tab` and having it be something else than the actual tab seems pretty slim though :smile:

Reply:
Is there any way we can check if it's truthy before it's set the first time, and throw an explicit error if it is? To decrease the risk that an overwrite is missed and causes bugs?

Reply:
Yeah sure, throwing an error might not be necessary though... we could just ignore setting it

Reply:
Because we assume it's already set to the right thing? If that was the case, overwriting it shouldn't be a problem either, I think? I'm thinking of the potential case where someone built a component that takes some completely different input on a property called `tab` and that component is later placed into our tabs. Unless that component immediately fails because the prop was replaced, the bug could be easy to miss, since you're probably going to test that the tabs you just set up work as intended, but you might not thoroughly test the components you put in them, if they've always worked before… 🤔

Reply:
If you have a component like that you could always wrap it inside another component. 

We have documented the behavior in the docs so I think I'm leaning towards just having it the way it is now. It seems we need a lot of book keeping if we want to keep track of which components to set the tab on... :slightly_frowning_face:

Reply:
Yeah, exactly. I don't think we should track it. I just mean that if it's simple for us to check if there's something there, for example in `componentWillLoad`, and throw an error with a helpful message, then I think it would be helpful to do that.

If it's not easy to check in a place where we know we haven't set anything ourselves yet, then I think it's fine to just do it the way we do it now, overwriting if there's a conflict.